pkg/util/provider/driver/fake.go

```diff
@@ -25,20 +25,31 @@ import (
 	"github.com/gardener/machine-controller-manager/pkg/util/provider/machinecodes/status"
 )
 
+// VMs is the map to hold the VM data
+type VMs map[string]string
+
 // FakeDriver is a fake driver returned when none of the actual drivers match
 type FakeDriver struct {
 	VMExists       bool
 	ProviderID     string
 	NodeName       string
 	LastKnownState string
 	Err            error
+	fakeVMs        VMs
 }
 
 // NewFakeDriver returns a new fakedriver object
 func NewFakeDriver(fakeDriver *FakeDriver) Driver {
```

Comment:
Should't we create a new struct instance here? Otherwise what is the point of this func?

Reply:
Yes, Amshu. Creating a new instance according to the params passed to it, definitely makes more sense considering the func `NewFakeDriver` exists. I will do the changes.